examples/player/simid_player.js

```diff
@@ -609,14 +592,68 @@ class SimidPlayer {
     console.log('The creative has asked for the player to ping ' + requestedUrlArray);
   }
 
+  /**
+   * Called when video playback is complete.
+   * @private
+   */
+  videoComplete() {
+    this.simidProtocol.sendMessage(MediaMessage.ENDED);
+
+    if (this.requestedDuration_ == NO_REQUESTED_DURATION) {
+      this.stopAd(StopCode.MEDIA_PLAYBACK_COMPLETE);
+    }
+  }
+
+  /**
+   * Called when creative requests a change in duration of ad.
+   * @private
+   */
   onRequestChangeAdDuration(incomingMessage) {
-    if (this.requestedDuration_ != NO_REQUESTED_DURATION) {
-      // TODO: Support multiple change duration requests.
-      this.simidProtocol.reject(incomingMessage);
+    this.requestedDuration_  = incomingMessage.args['duration'];
+    //If requested duration is negative seconds, play entire ad
+    if (this.requestedDuration_ != UNLIMITED_DURATION && this.requestedDuration_< 0) {
+      this.sendLog("requested duration too short: " + this.requestedDuration_ + " seconds");
+      setTimeout(() => {
```

Comment:
I don't think this is what we want.

How about,
if (this.requestedDuration_ != UNLIMITED_DURATION && this.requestedDuration_< 0) {
  const errorMessage = {
        errorCode : PlayerErrorCode:UNSUPPORTED_TIME,
        message: 'A negative duration is not valid.'
      }
      this.simidProtocol.reject(incomingMessage, errorMessage);
  }

Reply:
okay, if we want this logic then I will change the stopAd code to creative requested